deeppavlov/skills/ecommerce_bot/bot.py

```diff
@@ -0,0 +1,241 @@
+                           weights=(1,), penalty=False) for item in self.ec_data]
+
+            score_feat = [bleu_advanced(self.preprocess.lemmas(item['feat_nlped']),
+                          self.preprocess.lemmas(self.preprocess.filter_nlp(query)),
+                          weights=(0.3, 0.7), penalty=False) for idx, item in enumerate(self.ec_data)]
+
+            scores = np.mean(
+                [score_feat, score_title], axis=0).tolist()
+
+            scores_title = [(score, -len(self.ec_data[idx]['Title']))
+                            for idx, score in enumerate(scores)]
+
+            raw_scores_ar = np.array(scores_title, dtype=[
+                ('x', 'float_'), ('y', 'int_')])
+
+            results_args = np.argsort(raw_scores_ar, order=('x', 'y'))[
+                ::-1].tolist()
+
+            results_args_sim = [
+                idx for idx in results_args if scores[idx] >= self.min_similarity]
+
+            log.debug(f"Items before similarity filtering {len(results_args)} and after {len(results_args_sim)} with th={self.min_similarity} "+
+                      f"the best one has score {scores[results_args[0]]} with title {self.ec_data[results_args[0]]['Title']}")
+
+            for key, value in state.items():
+                log.debug(f"Filtering for {key}:{value}")
+
+                if key == 'Price':
+                    price = value
+                    log.debug(f"Items before price filtering {len(results_args_sim)} with price {price}")
+                    results_args_sim = [idx for idx in results_args_sim
+                                        if self.preprocess.price(self.ec_data[idx]) >= price[0] and
+                                        self.preprocess.price(self.ec_data[idx]) <= price[1] and
+                                        self.preprocess.price(self.ec_data[idx]) != 0]
+                    log.debug(f"Items after price filtering {len(results_args_sim)}")
+
+                elif key in ['query', 'start', 'stop']:
+                    continue
+
+                else:
+                    results_args_sim = [idx for idx in results_args_sim
+                                        if key in self.ec_data[idx]
+                                        if self.ec_data[idx][key].lower() == value.lower()]
+
+            response = []
+            for idx in results_args_sim[start:stop]:
+                temp = copy.copy(self.ec_data[idx])
+                del temp['title_nlped']
+                del temp['feat_nlped']
+                response.append(temp)
+
+            confidence = [(score_title[idx], score_feat[idx])
+                          for idx in results_args_sim[start:stop]]
+            entropies = self._entropy_subquery(results_args_sim)
+            log.debug(f"Total number of relevant answers {len(results_args_sim)}")
+
+        return json.dumps(({'items': response, 'entropy': entropies, 'total':len(results_args_sim)}, confidence, state))
```

Comment:
`json.dumps()` returns string, but the method signature shows `Tuple[Dict[str, Dict[Any, Any]], List[float], Dict[Any, Any]]`